docs/builders/iso/xenserver-iso.html.markdown

```diff
@@ -73,6 +73,10 @@ each category, the available options are alphabetized and described.
   run `xe template-list`. Setting the correct value hints to XenServer how to
   optimize the virtual hardware to work best with that operating system.
 
+* `destroy_vifs` (boolean) - Whether to destroy VIFs on the VM prior to
+  exporting. Removing them may make the image more generic and reusable.
+  Default is `false`.
+
```

Comment:
@heindsight can you explain a scenario where having the VIFs remain hurts the templates reusability? I understand the general sentiment that it is unnecessary to include extra configuration, but I don't quite see where this could manifest to a concrete problem.

Reply:
Hello - I couldn't resist chiming in.

On the surface, it seems sloppy to leave a 10MB container that will never be used again. Or is there an example where it could be?  

What will happen if it is left in place and ends up inside a template. Will we get TWO of these things when a VM is deployed from the template, and if so - will the "current" or new one be used correctly for deployment?

Or instead will a new deployment just overwrite the existing one?

How will a Windows VM react to a snapshot request with a drive like this hanging around?

Additionally, it seems wasteful - I know we live in an age where we can be sloppy with diskspace because you can get 1TB  drive for $40 or so, so... who cares about soaking up 10MB on these things. But not all VMs run on cheap harddrives 

In principal, it would seem to be best if temporary drives were really temporary and cleaned up after the fact. Garbage collection seems to be a good practice, whether in terms of RAM or disk space.

Reply:
@Roche-Mike this PR adds an option to remove virtual network interfaces from the the VM template. It has nothing to do with any 10MB temporary drive.

Reply:
> On the surface, it seems sloppy to leave a 10MB container that will never be used again. Or is there an example where it could be?

@Roche-Mike as Heinrich mentioned, this should be unrelated to what you described. However, I am interested in what you are describing and it's something we should consider if the plugin is having the correct behavior. Can you please open a new issue with the relevant context so we can have a focused discussion on that behavior?